brave/browser-laptop
# Safe browsing api

Test Instructions:

- Open Brave and wait ~2-3 mins (needed to load data files for the first time)
- Verify that data files have been loaded by navigating to ` %DATA_DIR%/Safe Browsing` and check if some `.store` files are present.

Safe Browsing URLs:

1. Open:
- https://testsafebrowsing.appspot.com/s/phishing.html
- https://testsafebrowsing.appspot.com/s/malware.html
- https://testsafebrowsing.appspot.com/s/malware_in_iframe.html
- https://testsafebrowsing.appspot.com/s/unwanted.html
- https://testsafebrowsing.appspot.com/s/image_small.html
- https://testsafebrowsing.appspot.com/s/bad_css.html
- https://testsafebrowsing.appspot.com/s/bad_javascript.html

Verify if appropriate error pages are displayed when these URLs are opened.

Safe Browsing Downloads:

1. Open:
- https://testsafebrowsing.appspot.com/s/content.exe
- https://testsafebrowsing.appspot.com/s/badrep.exe
- https://testsafebrowsing.appspot.com/s/unknown.exe
- https://testsafebrowsing.appspot.com/s/pua.exe

Verify the downloads are blocked.

Enable/Disable Safe Browsing for all sites:

1. Navigate to "about:preferences" and Shields > Select `Basic Safe Browsing`
2. Navigate to any of the above URLs to confirm Google safe browsing is disabled.
3. Navigate to `freehacks.us` to verify `Basic Safe Browsing` still works.
4. Navigate to "about:preferences" and Shields > Select `Disable Safe Browsing`
5. Navigate to any of the `testsafebrowsing` URLs to verify Google Safe Browsing is disabled
6. Navigate to `freehacks.us` to verify `Basic Safe Browsing` works.

cc: @luixxiul - Test instructions updated. Let me know if you have any questions.
